ssh/tailssh/tailssh.go

```diff
@@ -541,9 +635,27 @@ func (c *conn) resolveTerminalAction(s ssh.Session) (*tailcfg.SSHAction, error)
 		if url == "" {
 			return nil, errors.New("reached Action that lacked Accept, Reject, and HoldAndDelegate")
 		}
+		awaitIntrOnce.Do(func() {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				buf := make([]byte, 1)
+				for {
+					n, err := cr.ReadContext(ctx, buf)
+					if err != nil {
+						return
+					}
+					if n > 0 && buf[0] == 0x03 { // Ctrl-C
+						s.Stderr().Write([]byte("Canceled.\r\n"))
+						s.Close()
```

Comment:
How does the caller of `resolveTerminalAction` find out that the session was closed?

Reply:
gets a context error from fetchSSHAction? 🤷‍♂️

Reply:
maybe call `cancel` explicitly here?

Reply:
fixed